src/common/transformations/tests/transpose_sinking/ts_general_test.cpp

```diff
@@ -412,6 +412,29 @@ TEST_F(TransformationTestsF, TSGeneralTestMultipleTypes) {
     manager.register_pass<TSGeneral>();
 }
 
+TEST_F(TransformationTestsF, TSGeneralCheckShapeOfConstFoldingDisabled) {
+    using namespace transpose_sinking::testing::general;
+    ov::Shape input_shape = {96, 40, 55};
+    ov::Shape reshape_shape = {1, 96, 40, 55};
+    ov::element::Type input_type = ov::element::f32;
+    {
+        auto X = std::make_shared<Parameter>(input_type, input_shape);
+        auto Shape = std::make_shared<Parameter>(input_type, reshape_shape);
+
+        auto order = std::make_shared<Constant>(ov::element::u64, ov::Shape{3}, ov::Shape{0, 2, 1});
+        auto transpose = std::make_shared<Transpose>(X, order);
+
+        auto shape_of = std::make_shared<ShapeOf>(Shape);
+        auto reshape = std::make_shared<Reshape>(transpose, shape_of, false);
+
+        auto ng_order1 = std::make_shared<Constant>(ov::element::u64, ov::Shape{4}, ov::Shape{0, 3, 1, 2});
+        auto transpose1 = std::make_shared<Transpose>(reshape, ng_order1);
+
+        function = std::make_shared<ov::Model>(transpose1, ov::ParameterVector{X, Shape});
+    }
+    manager.register_pass<TSGeneral>();
+}
+
```

Comment:
can we have a reference for the comparison?

Reply:
we don't need to create a reference here, the transformation won't affect the original network due to ShapeOf subgraph.
a copy of the original network will be used as a reference